sdk/openai/Azure.AI.OpenAI/src/Custom/SseReader.cs

```diff
@@ -0,0 +1,147 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.IO;
+using System.Threading.Tasks;
+
+namespace Azure.Core.Sse
+{
+    // SSE specification: https://html.spec.whatwg.org/multipage/server-sent-events.html#parsing-an-event-stream
+    public sealed class SseReader : IDisposable
```

Comment:
Does this need to be public?

Reply:
Certainly not. I'll swap this (and the moved `SseLine`) to `internal`.